kafka/src/main/java/org/axonframework/extensions/kafka/eventhandling/consumer/KafkaSubscriberBuilder.java

```diff
@@ -0,0 +1,99 @@
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Pattern;
+
+import static org.axonframework.common.BuilderUtils.assertNonNull;
+import static org.axonframework.common.BuilderUtils.assertThat;
+
+/**
+ * Used by {@link org.axonframework.extensions.kafka.eventhandling.consumer.streamable.StreamableKafkaMessageSource.Builder} and {@link org.axonframework.extensions.kafka.eventhandling.consumer.subscribable.SubscribableKafkaMessageSource.Builder}
+ * to provide a {@link KafkaSubscriber} to subscribe a {@link Consumer} to topic(s).
+ *
+ * @author Ben Kornmeier
+ * @since 4.8
+ */
+public abstract class KafkaSubscriberBuilder<T extends KafkaSubscriberBuilder<T>> {
+    protected KafkaSubscriber subscriber = new ListKafkaSubscriber(Collections.singletonList("Axon.Events"));
+
+    /**
+     * Allows methods defined in this class to return the concrete class for fluent api usage.
+     * @return the current instance of the T
+     */
+    protected abstract T self();
+
+    public KafkaSubscriber getSubscriber() {
+        return subscriber;
+    }
+
+    /**
+     * Set the Kafka {@code topics} to read {@link org.axonframework.eventhandling.EventMessage}s from. Defaults to
+     * {@code Axon.Events}.
+     *
+     * @param topics the Kafka {@code topics} to read {@link org.axonframework.eventhandling.EventMessage}s from
+     * @return the current Builder instance, for fluent interfacing
+     */
+    public T topics(List<String> topics) {
+        assertThat(topics, topicList -> Objects.nonNull(topicList) && !topicList.isEmpty(),
+                "The topics may not be null or empty");
+        this.subscriber = new ListKafkaSubscriber(topics);
+        return self();
+    }
+    /**
+     * Add a Kafka {@code topic} to read {@link org.axonframework.eventhandling.EventMessage}s from.
+     *
+     * @param topic the Kafka {@code topic} to add to the list of topics
+     * @return the current Builder instance, for fluent interfacing
+     */
+    public T addTopic(String topic) {
+        assertThat(topic, name -> Objects.nonNull(name) && !"".equals(name), "The topic may not be null or empty");
+        if (isListBasedSubscription()) {
+            ((ListKafkaSubscriber) subscriber).addTopic(topic);
+        } else {
+            throw new IllegalStateException("Cannot add topic to a pattern subscriber");
+        }
+        return self();
+    }
+    /**
+     * Set the Kafka {@code pattern} to read {@link org.axonframework.eventhandling.EventMessage}s from.
+     *
+     * @param pattern the Kafka {@code pattern} to read {@link org.axonframework.eventhandling.EventMessage}s from
+     * @return the current Builder instance, for fluent interfacing
+     */
+    public T topicPattern(Pattern pattern) {
+        assertNonNull(pattern, "The pattern may not be null");
+        this.subscriber = new PatternKafkaSubscriber(pattern);
+        return self();
+    }
+
+    /**
+     * Return the type of subscription being used.
```

Comment:
I think the JavaDoc is a little off here. It returns if the topic subscriber is a `ListKafkaSubscriber`, not necessarily "the type."
If there'd be a third instance of the `KafkaSubscriber`/`TopicSubscriber`, for example, the JavaDoc wouldn't hold anymore.

Reply:
Ah, actually, since it's `private`, it would be fine to drop the JavaDoc altogether. I think the name of this private method is clear enough to explain to us what the intent is.